packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1945,7 +1947,33 @@ fn run_experiment(
             v8::V8::initialize_platform(platform);
             v8::V8::initialize();
 
-            let mut isolate = v8::Isolate::new(Default::default());
+            let mut create_params = v8::Isolate::create_params();
+
+            create_params = match (
+                init_msg.js_runner_initial_heap_constraint,
+                init_msg.js_runner_max_heap_size,
+            ) {
+                (None, None) => create_params,
+                (None, Some(js_runner_max_heap_size)) => {
+                    // 0 is V8's default.
+                    create_params.heap_limits(0, js_runner_max_heap_size * MB)
+                }
+                (Some(js_runner_initial_heap_constraint), None) => {
+                    // 0 is V8's default.
+                    create_params.heap_limits(js_runner_initial_heap_constraint * MB, 0)
+                }
+                (Some(js_runner_initial_heap_constraint), Some(js_runner_max_heap_size)) => {
+                    create_params.heap_limits(js_runner_initial_heap_constraint * MB, js_runner_max_heap_size * MB)
+                }
+            };
+
+            let mut isolate = v8::Isolate::new(create_params);
```

Comment:
We could simply do
```suggestion
            let initial_heap_constraint = init_msg.js_runner_initial_heap_constraint.unwrap_or_default();
            let max_heap_size = init_msg.js_runner_max_heap_size.unwrap_or_default();
            
            let mut isolate = v8::Isolate::new(
                v8::Isolate::create_params()
                    .heap_limits(initial_heap_constraint * MB, max_heap_size * MB
            ));
```

Or whatever `rustfmt` proposes.

Reply:
Simplified in https://github.com/hashintel/hash/pull/446/commits/2218e007c3e1d623dc3b7ffecbbd40958d8b466c

Reply:
@TimDiekmann I'm assuming you're happy with this? I think it's the last unresolved comment thread

Reply:
Yes, thanks!